File: app/src/main/java/com/example/abhishek/workoutdiary/MainActivity.java
package com.example.abhishek.workoutdiary;

import android.app.Activity;
import android.app.AlertDialog;
import android.app.FragmentManager;
import android.content.Context;
import android.content.DialogInterface;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;
import android.support.annotation.NonNull;
import android.support.design.widget.FloatingActionButton;
import android.support.design.widget.NavigationView;
import android.support.v4.app.Fragment;
import android.support.v4.content.ContextCompat;
import android.support.v4.widget.DrawerLayout;
import android.support.v7.app.ActionBarDrawerToggle;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.text.InputType;
import android.text.Layout;
import android.text.SpannableString;
import android.text.style.AlignmentSpan;
import android.text.style.ForegroundColorSpan;
import android.util.DisplayMetrics;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.EditText;
import android.widget.FrameLayout;
import android.widget.LinearLayout;
import android.widget.Spinner;
import android.widget.SpinnerAdapter;
import android.widget.Toast;

import java.util.List;

public class MainActivity extends AppCompatActivity implements NavigationView.OnNavigationItemSelectedListener{
    private List<String> exerciseList,cardioList;
    private customDatabase db;
    private cardioData data;
    private DrawerLayout drawerLayout;
    private ActionBarDrawerToggle toggle;
    private NavigationView nav,cardioNav;
    private boolean unit;

    /*
    initialize variables and set the drawer listener
    load the previous state of the activity if any
     */
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        drawerLayout=findViewById(R.id.drawerlayout);
        db=new customDatabase(this);
        data=new cardioData(this);
        exerciseList=db.getExercises();
        cardioList=data.getExerciseNames();
        nav=findViewById(R.id.nav);
        nav.setNavigationItemSelectedListener(this);
        cardioNav=findViewById(R.id.CardioNav);
        cardioNav.setNavigationItemSelectedListener(this);
        toggle=new ActionBarDrawerToggle(this,drawerLayout,R.string.drawer_open,R.string.drawer_close);
        drawerLayout.addDrawerListener(toggle);
        try {
            //noinspection ConstantConditions
            //to display the 3 lines icon in the home screen
            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
            getSupportActionBar().setHomeButtonEnabled(true);
        } catch (NullPointerException N) {
            Toast.makeText(MainActivity.this, "Oops! Something went wrong!", Toast.LENGTH_SHORT).show();
        }
        //if state wasn't saved before navigate to first exercise
        unit=getPreferences(Context.MODE_PRIVATE).getBoolean(getString(R.string.sharedPreferenceBoolean),false);
        if(nav.getMenu().size()>1){
            onNavigationItemSelected(nav.getMenu().getItem(1));
        }
        else if(cardioNav.getMenu().size()>1){
            onNavigationItemSelected(cardioNav.getMenu().getItem(1));
        }
        prepareFAB();
        for(String s:exerciseList){
            refreshMenu(s,false);
        }
        for(String s:cardioList)
            refreshMenu(s,true);
        //get previous state if any
        SharedPreferences preferences= PreferenceManager.getDefaultSharedPreferences(this);
        String name=preferences.getString(getString(R.string.saveState),"");
        if(name.length()>0){
            if(name.equalsIgnoreCase("graph")){
                Bundle b=new Bundle();
                b.putString("exercise",preferences.getString("graphSavedExercise",""));
                b.putString("type",preferences.getString("graphSavedType",""));
                graphFrag frag=new graphFrag();
                if(preferences.getString("graphSavedExercise","").length()>0) {
                    frag.setArguments(b);
                    getSupportFragmentManager().beginTransaction().replace(R.id.frameLayout, frag).
                            addToBackStack("Graph").commit();
                }
            }
            else if(name.equalsIgnoreCase("Settings")){
                FrameLayout f=findViewById(R.id.frameLayout);f.removeAllViews();
                getSupportFragmentManager().beginTransaction().replace(R.id.frameLayout,new Settings()).
                        addToBackStack("Settings").commit();
            }
            else{
                if(preferences.getString("fragmentType","").equals("strength")){
                    onNavigationItemSelected(nav.getMenu().getItem(exerciseList.indexOf(name)+1));
                }
                else{
                    if(cardioList.indexOf(name)!=-1)
                        onNavigationItemSelected(cardioNav.getMenu().getItem(cardioList.indexOf(name)+1));
                }
            }
        }
    }
    //to refresh the menu for the navigation view and repopulate the view
    /*
    input: exercise name and whether its cardio or not
    output: index of recently added exercise
     */
    private int refreshMenu(String s,boolean iscardio){
        Menu menu;int index;
        if(!iscardio) {
            menu = nav.getMenu();
            int len = menu.size();
            for (int i = 1; i <= len; i++) {
                menu.removeItem(i);
            }
            for (int i = 0; i < exerciseList.size(); i++) {
                exerciseAdded(exerciseList.get(i), i, false);
            }
            index=exerciseList.indexOf(s)+1;
        }
        else{
            menu=cardioNav.getMenu();
            int len=menu.size();
            for(int i=1;i<=len;i++){
                menu.removeItem(i);
            }
            for(int i=0;i<cardioList.size();i++){
                exerciseAdded(cardioList.get(i),i,true);
            }
            index=cardioList.indexOf(s)+1;
        }
        return index;
    }

    /*
    prepare the floating action button response in the navigation view
    sets up the listener to add a new exercise and navigates to the exercise fragment
     */
    private void prepareFAB() {
        FloatingActionButton fab = findViewById(R.id.newItem);
        fab.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                AlertDialog.Builder builder = new AlertDialog.Builder(MainActivity.this);
                final EditText e = new EditText(MainActivity.this);
                e.setSingleLine();
                e.setHint("Enter new Exercise");
                e.setInputType(InputType.TYPE_CLASS_TEXT | InputType.TYPE_TEXT_FLAG_CAP_WORDS);
                final Spinner sp=new Spinner(MainActivity.this);
                String vals[]=new String[]{"Strength","Cardio"};
                SpinnerAdapter spinnerAdapter=new ArrayAdapter<>(MainActivity.this,android.R.layout.simple_spinner_dropdown_item,
                        vals);
                sp.setAdapter(spinnerAdapter);
                LinearLayout l = new LinearLayout(MainActivity.this);
                l.setOrientation(LinearLayout.VERTICAL);
                l.addView(e);
                l.addView(sp);
                builder.setView(l);
                builder.setPositiveButton("Done", new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialog, int which) {
                        if (e.getText() == null) dialog.cancel();
                        String s = e.getText().toString();
                        String type=sp.getSelectedItem().toString();
                        if(type.equalsIgnoreCase("Strength")) {
                            if (!exerciseList.contains(s)) {
                                exerciseList.add(s);
                                int index=refreshMenu(s, false);
                                onNavigationItemSelected(nav.getMenu().getItem(index));
                            }
                            else
                                Toast.makeText(MainActivity.this,"Exercise is already present",Toast.LENGTH_LONG).show();
                        }
                        else{
                            if(!cardioList.contains(s)){
                                cardioList.add(s);
                                int index=refreshMenu(s,true);
                                onNavigationItemSelected(cardioNav.getMenu().getItem(index));
                            }
                            else
                                Toast.makeText(MainActivity.this,"Exercise is already present",Toast.LENGTH_LONG).show();
                        }
                    }
                });
                builder.create().show();
            }
        });
    }
    /*
    adds the exercise to the navigation view after making some ui changes to the text
    also adds dummy headers for the strength and cardio lists
    also adjusts the display height according to which list has more exercises
    input: exercise name, order number in the navigation view, cardio exercise or not
     */
    private void exerciseAdded(String s,int order, boolean iscardio){
        SpannableString string=new SpannableString(s);
        string.setSpan(new AlignmentSpan.Standard(Layout.Alignment.ALIGN_CENTER),0,s.length(),0);
        Menu menu=nav.getMenu(),menu1=cardioNav.getMenu();
        if(menu.size()==0){
            SpannableString strength=new SpannableString("Strength");
            strength.setSpan(new ForegroundColorSpan(ContextCompat.getColor(this,R.color.gray)),
                    0,strength.length(),0);
            menu.add(0,R.id.Strength,Menu.NONE, strength);
            menu.findItem(R.id.Strength).setCheckable(false);
        }
        if(menu1.size()==0){
            SpannableString cardio=new SpannableString("Cardio");
            cardio.setSpan(new ForegroundColorSpan(ContextCompat.getColor(this,R.color.gray)),
                    0,cardio.length(),0);
            menu1.add(0,R.id.Cardio,0,cardio);
            menu1.findItem(R.id.Cardio).setCheckable(false);
        }
        if(iscardio){
            if(cardioNav.getVisibility()==View.INVISIBLE)
                cardioNav.setVisibility(View.VISIBLE);
            menu1.add(0,order+1,order,string);
        }
        else{
            if(nav.getVisibility()==View.INVISIBLE)
                nav.setVisibility(View.VISIBLE);
            menu.add(0,order+1,order,string);
        }
        ViewGroup.LayoutParams p1,p2;
        p1=nav.getLayoutParams();p2=cardioNav.getLayoutParams();
        DisplayMetrics metrics=new DisplayMetrics();
        getWindowManager().getDefaultDisplay().getMetrics(metrics);
        int height=metrics.heightPixels-(int)(22.3/100*metrics.heightPixels);
        if(cardioNav.getMenu().size()>nav.getMenu().size()){
            p2.height=(int)((2.0/3)*height);
            p1.height=(int)((1.0/3)*height);
        }
        else if(nav.getMenu().size()>cardioNav.getMenu().size()){
            p1.height=(int)((2.0/3)*height);
            p2.height=(int)((1.0/3)*height);
        }
        else {
            p1.height=p2.height=height/2;
        }
        cardioNav.requestLayout();nav.requestLayout();
    }
    @Override
    protected void onPostCreate(Bundle savedInstanceState) {
        super.onPostCreate(savedInstanceState);
        toggle.syncState();
    }
    /*
    creates menu items for settings , graph and instructions
     */
    @Override
    public boolean onCreateOptionsMenu(Menu menu){
        MenuInflater inflater=getMenuInflater();
        inflater.inflate(R.menu.menu,menu);
        menu.add(0,R.id.workoutGraph,0,"See Progress");
        menu.add(0,R.id.instructions,0,"Instructions");
        menu.add(0,R.id.preference,0,"Settings");
        return true;
    }
    @Override
    public boolean onOptionsItemSelected(MenuItem item){
        return toggle.onOptionsItemSelected(item);
    }
    /*
    navigates to the exercise selected
    also adds the previous page into the backstack for the back button to be useful
    input: menu item selected: exercise, settings, graph, instructions
    output: whether it was selected
     */
    @Override
    public boolean onNavigationItemSelected(@NonNull MenuItem item) {
        final android.support.v4.app.FragmentManager manager=getSupportFragmentManager();
        if(item.getItemId()==R.id.Strength||item.getItemId()==R.id.Cardio){
            return true;
        }
        else if(item.getItemId()==R.id.preference){
            Fragment fragment=new Settings();
            manager.beginTransaction().replace(R.id.frameLayout,fragment,"Settings").addToBackStack("Settings").commit();
            return true;
        }
        setTitle(item.getTitle());
        FrameLayout f = findViewById(R.id.frameLayout);
        f.removeAllViews();Fragment fragment;
        String exercise=item.getTitle().toString();
        if(exerciseList.contains(exercise)) {
            fragment = new strengthFrag();
        }
        else
            fragment=new cardioFrag();
        Bundle b=new Bundle();
        b.putString("name",exercise);
        fragment.setArguments(b);
        manager.beginTransaction().replace(R.id.frameLayout,fragment,exercise).addToBackStack(
                exercise).commit();
        manager.addOnBackStackChangedListener(new android.support.v4.app.FragmentManager.OnBackStackChangedListener() {
            @Override
            public void onBackStackChanged() {
                int index=manager.getBackStackEntryCount()-1;
                if(index<0){
                    MainActivity.this.finishAndRemoveTask();
                }
                else{
                    String title=manager.getBackStackEntryAt(index).getName();
                    setTitle(title);
                }
            }
        });
        drawerLayout.closeDrawer(findViewById(R.id.navMain));
        return true;
    }
}

File: app/src/main/java/com/example/abhishek/workoutdiary/MainActivity.kt
package com.example.abhishek.workoutdiary

import android.app.AlertDialog
import android.content.Context
import android.os.Bundle
import android.text.InputType
import android.text.Layout
import android.text.SpannableString
import android.text.style.AlignmentSpan
import android.text.style.ForegroundColorSpan
import android.util.DisplayMetrics
import android.view.Menu
import android.view.MenuItem
import android.view.View
import android.view.ViewGroup
import android.widget.*
import androidx.appcompat.app.ActionBarDrawerToggle
import androidx.appcompat.app.AppCompatActivity
import androidx.core.content.ContextCompat
import androidx.core.view.GravityCompat
import androidx.drawerlayout.widget.DrawerLayout
import androidx.fragment.app.Fragment
import androidx.preference.PreferenceManager
import com.example.abhishek.workoutdiary.Interfaces.DrawerListener
import com.google.android.material.floatingactionbutton.FloatingActionButton
import com.google.android.material.navigation.NavigationView

class MainActivity : AppCompatActivity(), NavigationView.OnNavigationItemSelectedListener, DrawerListener {
    private var exerciseList = mutableListOf<String>()
    private var cardioList = mutableListOf<String>()
    private var db: customDatabase = customDatabase(this)
    private var data = cardioData(this)
    private lateinit var drawerLayout: DrawerLayout
    private lateinit var toggle: ActionBarDrawerToggle
    private lateinit var nav: NavigationView
    private lateinit var cardioNav: NavigationView
    private var unit = false
    /*
    initialize variables and set the drawer listener
    load the previous state of the activity if any
     */
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        drawerLayout = findViewById(R.id.drawerlayout)
        //        Rect r = drawerLayout.cop
//        drawerLayout.setSystemGestureExclusionRects(rr);
        exerciseList = db.exercises
        cardioList = data.exerciseNames
        nav = findViewById(R.id.nav)
        nav.setNavigationItemSelectedListener(this)
        cardioNav = findViewById(R.id.CardioNav)
        cardioNav.setNavigationItemSelectedListener(this)
        toggle = ActionBarDrawerToggle(this, drawerLayout, R.string.drawer_open, R.string.drawer_close)
        drawerLayout.addDrawerListener(toggle)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        supportActionBar?.setHomeButtonEnabled(true)
        //if state wasn't saved before navigate to first exercise
        unit = getPreferences(Context.MODE_PRIVATE).getBoolean(getString(R.string.sharedPreferenceBoolean), false)
        if (nav.menu.size() > 1) {
            onNavigationItemSelected(nav.menu.getItem(1))
        } else if (cardioNav.menu.size() > 1) {
            onNavigationItemSelected(cardioNav.menu.getItem(1))
        }
        prepareFAB()
        val drawerOpenButton = findViewById<FloatingActionButton>(R.id.open_drawer)
        drawerOpenButton.setOnClickListener { onDrawerOpen() }
        for (s in exerciseList) {
            refreshMenu(s, false)
        }
        for (s in cardioList) refreshMenu(s, true)
        //get previous state if any
        val preferences = PreferenceManager.getDefaultSharedPreferences(this)
        val name = preferences.getString(getString(R.string.saveState), "")
        if (name != null && name.isNotEmpty()) {
            if (name.equals("graph", ignoreCase = true)) {
                val b = Bundle()
                b.putString("exercise", preferences.getString("graphSavedExercise", ""))
                b.putString("type", preferences.getString("graphSavedType", ""))
                val frag = graphFrag()
                val graphSavedExercise = preferences.getString("graphSavedExercise", "")
                if (graphSavedExercise != null && graphSavedExercise.isNotEmpty()) {
                    frag.arguments = b
                    supportFragmentManager.beginTransaction().replace(R.id.frameLayout, frag).addToBackStack("Graph").commit()
                }
            } else if (name.equals("Settings", ignoreCase = true)) {
                val f = findViewById<FrameLayout>(R.id.frameLayout)
                f.removeAllViews()
                supportFragmentManager.beginTransaction().replace(R.id.frameLayout, Settings()).addToBackStack("Settings").commit()
            } else if (exerciseList.size > 0 || cardioList.size > 0) {
                if (preferences.getString("fragmentType", "") == "strength") {
                    if (exerciseList.contains(name)) onNavigationItemSelected(nav.menu.getItem(exerciseList.indexOf(name) + 1)) else onNavigationItemSelected(nav.menu.getItem(1))
                } else {
                    if (cardioList.contains(name)) onNavigationItemSelected(cardioNav.menu.getItem(cardioList.indexOf(name) + 1)) else onNavigationItemSelected(cardioNav.menu.getItem(1))
                }
            }
        }
    }

    //to refresh the menu for the navigation view and repopulate the view
/*
    input: exercise name and whether its cardio or not
    output: index of recently added exercise
     */
    private fun refreshMenu(s: String, iscardio: Boolean): Int {
        val menu: Menu
        val index: Int
        if (!iscardio) {
            menu = nav.menu
            val len = menu.size()
            for (i in 1..len) {
                menu.removeItem(i)
            }
            for (i in exerciseList.indices) {
                exerciseAdded(exerciseList[i], i, false)
            }
            index = exerciseList.indexOf(s) + 1
        } else {
            menu = cardioNav.menu
            val len = menu.size()
            for (i in 1..len) {
                menu.removeItem(i)
            }
            for (i in cardioList.indices) {
                exerciseAdded(cardioList[i], i, true)
            }
            index = cardioList.indexOf(s) + 1
        }
        return index
    }

    /*
    prepare the floating action button response in the navigation view
    sets up the listener to add a new exercise and navigates to the exercise fragment
     */
    private fun prepareFAB() {
        val fab = findViewById<FloatingActionButton>(R.id.newItem)
        fab.setOnClickListener {
            val builder = AlertDialog.Builder(this@MainActivity)
            val e = EditText(this@MainActivity)
            e.setSingleLine()
            e.hint = "Enter new Exercise"
            e.inputType = InputType.TYPE_CLASS_TEXT or InputType.TYPE_TEXT_FLAG_CAP_WORDS
            val sp = Spinner(this@MainActivity)
            val vals = arrayOf("Strength", "Cardio")
            val spinnerAdapter: SpinnerAdapter = ArrayAdapter(this@MainActivity, android.R.layout.simple_spinner_dropdown_item,
                    vals)
            sp.adapter = spinnerAdapter
            val l = LinearLayout(this@MainActivity)
            l.orientation = LinearLayout.VERTICAL
            l.addView(e)
            l.addView(sp)
            builder.setView(l)
            builder.setPositiveButton("Done") { dialog, _ ->
                if (e.text == null) dialog.cancel()
                val s = e.text.toString()
                val type = sp.selectedItem.toString()
                if (type.equals("Strength", ignoreCase = true)) {
                    if (!exerciseList.contains(s)) {
                        exerciseList.add(s)
                        val index = refreshMenu(s, false)
                        onNavigationItemSelected(nav.menu.getItem(index))
                    } else Toast.makeText(this@MainActivity, "Exercise is already present", Toast.LENGTH_LONG).show()
                } else {
                    if (!cardioList.contains(s)) {
                        cardioList.add(s)
                        val index = refreshMenu(s, true)
                        onNavigationItemSelected(cardioNav.menu.getItem(index))
                    } else Toast.makeText(this@MainActivity, "Exercise is already present", Toast.LENGTH_LONG).show()
                }
            }
            builder.create().show()
        }
    }

    /*
    adds the exercise to the navigation view after making some ui changes to the text
    also adds dummy headers for the strength and cardio lists
    also adjusts the display height according to which list has more exercises
    input: exercise name, order number in the navigation view, cardio exercise or not
     */
    private fun exerciseAdded(s: String, order: Int, iscardio: Boolean) {
        val string = SpannableString(s)
        string.setSpan(AlignmentSpan.Standard(Layout.Alignment.ALIGN_CENTER), 0, s.length, 0)
        val menu = nav.menu
        val menu1 = cardioNav.menu
        if (menu.size() == 0) {
            val strength = SpannableString("Strength")
            strength.setSpan(ForegroundColorSpan(ContextCompat.getColor(this, R.color.gray)),
                    0, strength.length, 0)
            menu.add(0, R.id.Strength, Menu.NONE, strength)
            menu.findItem(R.id.Strength).isCheckable = false
        }
        if (menu1.size() == 0) {
            val cardio = SpannableString("Cardio")
            cardio.setSpan(ForegroundColorSpan(ContextCompat.getColor(this, R.color.gray)),
                    0, cardio.length, 0)
            menu1.add(0, R.id.Cardio, 0, cardio)
            menu1.findItem(R.id.Cardio).isCheckable = false
        }
        if (iscardio) {
            if (cardioNav.visibility == View.INVISIBLE) cardioNav.visibility = View.VISIBLE
            menu1.add(0, order + 1, order, string)
        } else {
            if (nav.visibility == View.INVISIBLE) nav.visibility = View.VISIBLE
            menu.add(0, order + 1, order, string)
        }
        val p1: ViewGroup.LayoutParams = nav.layoutParams
        val p2: ViewGroup.LayoutParams = cardioNav.layoutParams
        val metrics = DisplayMetrics()
        windowManager.defaultDisplay.getMetrics(metrics)
        val height = metrics.heightPixels - (22.3 / 100 * metrics.heightPixels).toInt()
        val menuSizeDiff = cardioNav.menu.size() - nav.menu.size()
        when  {
            menuSizeDiff > 0 -> {
                p2.height = (2.0 / 3 * height).toInt()
                p1.height = (1.0 / 3 * height).toInt()
            }
            menuSizeDiff < 0 -> {
                p1.height = (2.0 / 3 * height).toInt()
                p2.height = (1.0 / 3 * height).toInt()
            }
            else -> {
                p2.height = height / 2
                p1.height = p2.height
            }
        }
        cardioNav.requestLayout()
        nav.requestLayout()
    }

    override fun onPostCreate(savedInstanceState: Bundle?) {
        super.onPostCreate(savedInstanceState)
        toggle.syncState()
    }

    /*
    creates menu items for settings , graph and instructions
     */
    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        if (exerciseList.size == 0 && cardioList.size == 0) return false
        val inflater = menuInflater
        inflater.inflate(R.menu.menu, menu)
        menu.add(0, R.id.workoutGraph, 0, "See Progress")
        menu.add(0, R.id.instructions, 0, "Instructions")
        menu.add(0, R.id.preference, 0, "Settings")
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return toggle.onOptionsItemSelected(item)
    }

    /*
    navigates to the exercise selected
    also adds the previous page into the backstack for the back button to be useful
    input: menu item selected: exercise, settings, graph, instructions
    output: whether it was selected
     */
    override fun onNavigationItemSelected(item: MenuItem): Boolean {
        val manager = supportFragmentManager
        if (item.itemId == R.id.Strength || item.itemId == R.id.Cardio) {
            return true
        } else if (item.itemId == R.id.preference) {
            val fragment: Fragment = Settings()
            manager.beginTransaction().replace(R.id.frameLayout, fragment, "Settings").addToBackStack("Settings").commit()
            return true
        }
        val f = findViewById<FrameLayout>(R.id.frameLayout)
        f.removeAllViews()
        val fragment: Fragment
        val exercise = item.title.toString()
        fragment = if (exerciseList.contains(exercise)) {
            strengthFrag()
        } else cardioFrag()
        val b = Bundle()
        b.putString("name", exercise)
        fragment.arguments = b
        manager.beginTransaction().replace(R.id.frameLayout, fragment, exercise).addToBackStack(
                exercise).commit()
        manager.addOnBackStackChangedListener {
            val index = manager.backStackEntryCount - 1
            if (index < 0) {
                finishAndRemoveTask()
            } else {
                val title = manager.getBackStackEntryAt(index).name
                setTitle(title)
            }
        }
        drawerLayout.closeDrawer(findViewById<View>(R.id.navMain))
        return true
    }

    override fun onDrawerOpen() {
        drawerLayout.openDrawer(GravityCompat.START)
    }
}